src/Temporalio/Client/ITemporalClient.Workflow.cs

```diff
@@ -183,5 +183,24 @@ public IAsyncEnumerable<WorkflowExecution> ListWorkflowsAsync(
         /// <seealso href="https://docs.temporal.io/visibility">Visibility docs.</seealso>
         public Task<WorkflowExecutionCount> CountWorkflowsAsync(
             string query, WorkflowCountOptions? options = null);
+
+        /// <summary>
+        /// List workflows with the given query using manual paging.
+        /// </summary>
+        /// <param name="query">
+        /// Query to use for filtering. Subsequent pages must have the same query as the initial call.
+        /// </param>
+        /// <param name="nextPageToken">
+        /// Set to null for the initial call to retrieve the first page.
+        /// Set to <see cref="ListWorkflowsPage.NextPageToken"/> returned by a previous call to retrieve the next page.
+        /// </param>
+        /// <param name="options">Options for the list call.</param>
+        /// <returns>
+        /// A single page of a list of workflows.
+        /// Repeat the call using <see cref="ListWorkflowsPage.NextPageToken"/> to get more pages.
+        /// </returns>
+        /// <seealso href="https://docs.temporal.io/visibility">Visibility docs.</seealso>
+        Task<ListWorkflowsPage> GetListWorkflowsPageAsync(
```

Comment:
```suggestion
        Task<WorkflowListPage> FetchWorkflowListPageAsync(
```

would recommend this for a few reasons, namely because a "workflow list page" makes more sense than "list workflows page" IMO, and we don't really want to use getters for remote RPC calls. Could also just add `Paginated` to the existing call like so:

```suggestion
        Task<WorkflowListPage> ListWorkflowsPaginatedAsync(
```

That is also fairly clean I think